### ISSUE TITLE
[Task]: Search Initial Load - Show Non-Results Content Upon First Load (Suspense Boundary?)

### ISSUE THREAD

**Issue by acouch** [HHS/simpler-grants-gov#1804]
### Summary

Currently, the initial results loading looks like:

![image](https://github.com/HHS/simpler-grants-gov/assets/512243/878bc302-8c12-4faa-bcc6-67af60c366d4)

This task is to update the suspense boundary and possibly component layout/loading in order to deliver the contents of the page before the initial results are returned.

### Acceptance criteria

- [ ] Page components load before initial results are returned

**Comment by rylew1** [comment]
@acouch  The `<Suspense>` boundary I believe needs to be defined in the server page component (`/app/searhc/page`) - I'm thinking this might require pulling a bunch of client components up into server components. We did not design it out the gate for streaming well. 

If you look at the current setup - the components that could be streamed in instantly (without having to wait for the data fetch to be completed) -`SearchFilterAccordion`, `SearchBar` etc.. - these are all client components currently because they require interactivity (ie... you can type in them or click them). 

If we take the case of the `SearchBar` -  since it's fairly straightforward.
- All the things surrounding the `<input>` tag can be pulled into a server component `SearchBar` 
- The input itself needs to remain a client component . So just move it to a new/renamed `SearchInput` component. I believe this `SearchInput` component would be a child of the server component `SearchBar`
- `SearchBar` could then be moved to `app/search/page.tsx` 

So basically, separate be static and non-interactive into server components, then split out. Do this for most of the components. We could test that this actually works with suspense on a smaller scale first though (maybe there should be a ticket for that)


Current component hierarchy:

![image](https://github.com/HHS/simpler-grants-gov/assets/93001277/738058ed-16d3-4164-be6d-6e30ba1fe527)



 There may be an easier solution, and we can spin on it for a bit and maybe tackle this after user feedback.

**Comment by acouch** [comment]
Closed in [#2067](https://github.com/HHS/simpler-grants-gov/issues/2067)
